sway-core/src/semantic_analysis/ast_node/declaration/abi.rs

```diff
@@ -67,6 +67,16 @@ impl TypedAbiDeclaration {
             warnings,
             errors
         );
+        for typed_fn in &interface_surface {
+            for param in &typed_fn.parameters {
+                if param.is_reference {
+                    errors.push(CompileError::RefMutableNotAllowedInContractAbi {
+                        param_name: param.name.clone(),
+                    })
+                }
```

Comment:
How about 
```suggestion
                if param.is_reference && param.is_mutable {
                    errors.push(CompileError::RefMutableNotAllowedInContractAbi {
                        param_name: param.name.clone(),
                    })
                }
```
just for readability and if we decide to support only `mut` params or only `ref` params in the future?